lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -23,19 +24,35 @@ export default class DocumentComposer {
 
     const document = didState.document as DocumentModel;
 
-    // Only transform `publicKeys` to  `publickey` if the array is present.
-    let publicKeys;
+    // Only populate `publicKey` if general usage exists.
+    // Only populate `authentication` if auth usage exists.
```

Comment:
This implies we are sorta supporting https://w3c-ccg.github.io/ld-proofs/#proof-purpose, but not being explicit about it... probably better to be explicit.

Reply:
I am following the spec Daniel created, don't know if that is sufficient.